### PR TITLE
Update ec2_ami_search module lto be able to search for different amis on ec2

### PR DESCRIPTION
**Pull request by haad**
Change ec2_ami_search  to be able to search amazon for given images based on tags, name, uuid, or any other option described in describe images manual page.

```
- name: Search for ami
  ami_search:
   region: us-east-1
   owner: 'self'
   ami_tags: {"ami_uuid": '{{build_ami_id}}' }
  register: image
```

### CODE REVIEW

**Comment by haad** [comment]
Hmm any reason to not merge it ?

**Comment by bcoca** [comment]
I'm thinking to either rename this to a _facts module or to redo it as a lookup plugin.

File: cloud/amazon/ec2_ami_search.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/python
 #
-# (c) 2013, Nimbis Services
```
Comment (bcoca) [inline cloud/amazon/ec2_ami_search.py:3]: append your copyright, but don't remove others

File: cloud/amazon/ec2_ami_search.py [hunk 2]
```diff
@@ -16,185 +16,102 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
-
 DOCUMENTATION = '''
 ---
-module: ec2_ami_search
-short_description: Retrieve AWS AMI information for a given operating system.
-version_added: "1.6"
+module: ami_search
```
Comment (bcoca) [inline cloud/amazon/ec2_ami_search.py:21]: wrong module name?

File: cloud/amazon/ec2_ami_search.py [hunk 3]
```diff
@@ -16,185 +16,102 @@
...
+short_description: This module searchs given ami store for a ami with gitven attributes
 description:
-  - Look up the most recent AMI on AWS for a given operating system.
-  - Returns C(ami), C(aki), C(ari), C(serial), C(tag)
-  - If there is no AKI or ARI associated with an image, these will be C(null).
-  - Only supports images from cloud-images.ubuntu.com
-  - 'Example output: C({"ami": "ami-69f5a900", "changed": false, "aki": "aki-88aa75e1", "tag": "release", "ari": null, "serial": "20131024"})'
-version_added: "1.6"
+Module can search amis owned by different users. It will accept following parameters for atribute owner
```
Comment (bcoca) [inline cloud/amazon/ec2_ami_search.py:24]: description is not a list, it should be, test with ansible-doc when in doubt

**Comment by bcoca** [comment]
many changes are backwards incompatible, which would make me push for a new module name and deprecating the old one so as to not break existing playbooks

**Comment by bcoca** [comment]
closing this in favor of #905

**Pull request closed without merging**